feat: JoinFamilyGroupNameViewController 화면 구현

Do-hyun-Kim opened the pull request:
## 작업 내용 🧑‍💻

- JoinFamilyGroupNameViewController 화면 및 Reactor 추가
- 가족방 이름 변경 UseCase Interface `UpdateFamilyNameUseCaseProtocol` 추가
- 내 가족 정보 조회 UseCase Interface `FetchFamilyGroupInfoUseCaseProtocol`추가
- FamilyManagementViewController JoinFamilyGroupNameViewController 화면 전환 로직 추가

## 변경 로직 ⚒️
-  FamilyManagementViewControlle에 `JoinFamilyGroupNameViewController` 추가 했습니다.
- `FamailyRepository` 에 `가족방 이름 변경`, `내 가족 정보 조회` 관련  메서드 추가
- `Family DataMapping`  내부에 `가족방 이름 변경`,  `내 개족 정보 조회` 관련 DTO 추가

## 화면 (Optional)
| 가족방 이름 초기 설정 화면 | 가족방 이름 설정 이후 화면 |
|--------|--------|
| <img src="https://github.com/user-attachments/assets/b8c1b7c9-7160-4e1d-ab73-25bbb6d02f15" width="200"/> | <img src="https://github.com/user-attachments/assets/8bf4cb13-b66a-4955-95af-5a66a0bdff1e" width="200"/> |


## 테스트 케이스 ✅

- [ ] 가족방 네임 설정 API 호출 정상적으로 호출 되는지 확인 하기
- [ ] 각 디바이스 환경에서 최종적으로 가족방 이름 수정한 사용자 프로필을 호출하는지 확인 하기

Do-hyun-Kim committed:
feat: JoinFamilyGroupNameViewController, UseCase, Repository 추가
- JoinFamilyGroupNameViewControllerWrapper 추가
- FamilyAPIs JoinFamilyGroupName 변경 API 추가
- FamilyAPIWorker FamilyName 변경 API 로직 추가
- Data Layer JoinFamilyGroupNameRequestDTO, ResponseDTO 추가
- Domain Layer JoinFamilyGroupName Request, Entity 추가

Do-hyun-Kim committed:
fix: JoinFamilyGroupNameResponseDTO, JoinFamilyGroupNameRequestDTO 제거
- FamilyAPIs, APIWorker 가족방 이름 설정 API 로직 제거
- JoinFamilyGroupNameViewController UI 수정
- JoinFamilyGroupNameViewReactor Mutation, State 제거

Do-hyun-Kim committed:
Merge branch 'develop' of https://github.com/depromeet/14th-team5-iOS into feat/#594-add-family-room-ui

* 'develop' of https://github.com/depromeet/14th-team5-iOS: (23 commits)
  feat: 멤버ID, 이름을 옵셔널로 반환하도록 코드 수정 (#600)
  feat: FetchFamilyIdUseCase 구현 (#603)
  feat: 가족이름변경API통신, UpdateFamilyNameUseCase 구현 (#603)
  feat: MyRepository, MyUseCase 구현 (#600)
  docs: BBAlert, BBToast에 문서 주석 추가 (#598)
  feat: BBToastStyle 구현 (#598)
  feat: BBAlert의 버튼 동작 구현 (#598)
  feat: BBButton 하이라이트 효과 구현 (#598)
  feat: BBAlertStyle 구현 (#598)
  move: BibbiAlertBuilder Deprecated 처리 (#598)
  feat: TextAlertView 구현 (#598)
  feat: BBAlertButtonLayout 구현 (#598)
  feat: ImageAlertView 구현 (#598)
  feat: BBAlert 구현 (#598)
  feat: ToastButton 구현 (#591)
  feat: 구 ToastMessage 코드 Deprecated 처리 (#591)
  feat: BBToast 구현 (#591)
  move: BibbiToastMessageView 파일 이동 (#591)
  feat: BBNavigationBar의 set 메서드 이름 수정 (#587)
  move: BibbiNavigationBarView 폴더 이동 (#587)
  ...

# Conflicts:
#	14th-team5-iOS/Data/Sources/APIs/Family/FamilyAPI/FamilyAPIWorker.swift

Do-hyun-Kim committed:
feat: JoinFamilyGroupNameViewWrapper return 구문 수정
- FamilyManagementViewController tableEditButton UI Configure or JoinFamilyGroupNameViewController 화면 전환 코드 추가
- JoinFamilyGroupNameViewReactor memberUseCase, updateFamilyNameUseCase, fetchFamilyEditerUseCase 추가
- JoinFamilyGroupNameViewReactor viewDidLoad, setFamilyGroupNameEditorItem, setUpdateFamilyNameItem Mutation 추가
- JoinFamilyGroupNameViewReactor FamilyGroupNameEditorId에 따라 UIConfigure
- JoinFamilyGroupNameViewController Binding 코드 추가
- FamilyAPIWorker APIHeader 값 추가
- FamilyNameResponseDTO 오타 수정

rlarjsdn3 commented on 14th-team5-iOS/App/Sources/Application/Navigator/Wrapper/JoinFamily/JoinFamilyGroupNameViewControllerWrapper.swift:
```diff
@@ -0,0 +1,33 @@
+import Foundation
+
+
+final class JoinFamilyGroupNameViewControllerWrapper: BaseWrapper {
```
JoinFamilyGroupNameViewController 보단 그냥 FamilyNameSettingViewController 어떠세여? 다른 가족에 새로 들어가는 화면은 아니고 그냥 설정하는 화면이니까요

rlarjsdn3 replied:
그리고 해당 뷰컨트롤러와 리액터 파일들 그냥 FamilyManagement 폴더로 이동하는 게 어때요? 지금 당장은 FamilyManagement 화면에서 가족 이름 세팅 화면으로 이동되니까

Do-hyun-Kim replied:
- FamilyNameSettingViewController가 더 명확한 네이밍일 것 같네요 그렇게 수정하겠습니다.
- 넵 해당 폴더로 이동 해놓을 게영!!

rlarjsdn3 commented on 14th-team5-iOS/App/Sources/Presentation/JoinFamily/Views/JoinFamilyGroupEdierView.swift:
```diff
@@ -0,0 +1,68 @@
+
+
+final class JoinFamilyGroupEdierView: UIView {
+    
```
Edier 오타

Do-hyun-Kim replied:
아 저부분을 확인 못했네여 이부분도 수정해서 올리겠습니다.

rlarjsdn3 commented on 14th-team5-iOS/App/Sources/Presentation/JoinFamily/ViewController/JoinFamilyGroupNameViewController.swift:
```diff
@@ -0,0 +1,209 @@
+import ReactorKit
+
+final class JoinFamilyGroupNameViewController: BaseViewController<JoinFamilyGroupNameViewReactor> {
+    
```
BBNavigationController로 해주세염

Do-hyun-Kim replied:
넵!! :)

rlarjsdn3 commented on 14th-team5-iOS/App/Sources/Presentation/JoinFamily/ViewController/JoinFamilyGroupNameViewController.swift:
```diff
@@ -0,0 +1,209 @@
+    private let groupErrorLabel: BBLabel = BBLabel(.body1Regular, textColor: .warningRed)
+    private let groupEditerView: JoinFamilyGroupEdierView = JoinFamilyGroupEdierView()
+    
+    
```
피그마에 보면 ①처음 가족 이름 세팅하는 화면 ②이후 가족 이름 세팅하는 화면으로 나뉘는데, 나누어져 있는 건가요?

Do-hyun-Kim committed:
fix: FamilyNameSettingViewController, FamilyNameSettingViewReactor, FamilyNameSettingViewControllerWrapper 네이밍 수정
- FamilyNameSettingViewController, FamilyNameSettingViewReactor FamilyManagement 폴더로 디렉토리 위치 변경
- JoinFamilyGroupEditorView 오타 수정

Do-hyun-Kim committed:
Merge branch 'develop' of https://github.com/depromeet/14th-team5-iOS into feat/#594-add-family-room-ui

* 'develop' of https://github.com/depromeet/14th-team5-iOS: (21 commits)
  밤에 컨트리뷰터 뷰가 보이지 않는 문제를 해결합니다. (#624)
  rename method(#615)
  feat: add remove method(#615)
  feat: add userdefaults apps and family(#615)
  feat: CommentUserDefaults 구현 (#613)
  feat: App, My 및 FamilyUserDefaults 구현 (#613)
  fix: FamilyRepository에서 FamilyID를 제대로 못 불러오는 문제 수정 (#613)
  feat: BBProgressHUD에 airplaneWithTitle Style 구현 (#610)
  fix: BBProgressHUDSubView를 BBProgressHudStackView로 수정 (#610)
  feat: 구 LottieView 코드 Deprecated 처리 (#610)
  docs: 문서 주석 작성 (#610)
  feat: BBProgressHUDStyle 구현 (#610)
  feat: LottieProgressHUDView 구현 (#610)
  feat: BBLottieView 구현 (#610)
  feat: BBProgressHUD 구현, ButtonToastView 빌드 에러 수정 (#610)
  feat: BBNavigationBarButton 구현 (#606)
  feat: rename folder(#608)
  feat: modify view and rename(#608)
  fix: Constraints 충돌 에러 수정 (#606)
  feat: NavigationBar에 Rx+Ext 구현 (#606)
  ...

Do-hyun-Kim committed:
feat: Tuist infoPlist DarkMode 수정
- FamilyNameSettingViewReactor fetchFamilyGroupInfoUseCase 추가
- FamilyNameSettingViewReactor familyGroupInfo 통신 코드 로직 추가
- FamilyAPIs, APIWorker 가족 조회 API 추가
- 가족 조회 DTO, Entity 추가
- FetchFamilyGroupInfoUseCase 추가

Do-hyun-Kim committed:
fix: UpdateFamilyNameRequestDTO, FamilyNameRequestDTO FamilyName, FamilyNameEditorId Optional 처리로 수정
- JoinFamilyGroupEditorView profileNameLabel Font 수정

Do-hyun-Kim committed:
feat: FamilyNameSettingViewReactor 가족방 이름 초기화 로직 추가

Do-hyun-Kim committed:
feat: FamilyNameSettingViewController BBAlert 추가 및 delegate 채택

rlarjsdn3 commented:
수고하셨슴돠
